network/server.go

```diff
@@ -327,26 +329,61 @@ func (s *Server) setupBootnodes() error {
 }
 
 // keepAliveMinimumPeerConnections will attempt to make new connections
-// if the active peer count is lesser than the specified limit.
+// if the active peer count is lesser than the specified limit. And will attempt
+// to ensure that all bootnodes are connected to
 func (s *Server) keepAliveMinimumPeerConnections() {
+	bootNodes := s.bootnodes.getBootnodes()
+	ourId := s.host.ID()
+
+	// remove ourselves from the list of bootNodes if present
+	for i, bootNode := range bootNodes {
+		if bootNode.ID == ourId {
+			bootNodes = append(bootNodes[:i], bootNodes[i+1:]...)
```

Comment:
Double check this, but not necessary looping through the entire bootNodes slice if we find the "local node" among the bootnodes:

```suggestion
		bootNodes = append(bootNodes[:i], bootNodes[i+1:]...)
        break
```